builtin/nomad/jobspec/releaser.go

```diff
@@ -0,0 +1,598 @@
+
+If your Nomad deployment is configured to use canaries, this releaser plugin lets
+you promote (or fail) the canary deployment. You may also target specific task
+groups within your job for promotion, if you have multiple task groups in your canary
+deployment.
+
+-> **Note:** Using the ` + "`-prune=false`" + ` flag is recommended for this releaser. By default,
+Waypoint prunes and destroys all unreleased deployments and keeps only one previous
+deployment. Therefore, if ` + "`-prune=false`" + ` is not set, Waypoint may delete
+your job via "pruning" a previous version. See [deployment pruning](/docs/lifecycle/release#deployment-pruning)
+for more information.
+
+### Release URL
+
+If you want the URL of the release of your deployment to be published in Waypoint,
+you must set the meta 'waypoint.hashicorp.com/release_url' in your jobspec. The
+value specified in this meta field will be published as the release URL for your
+application. In the future, this may source from Consul.
+
+`)
+
+	doc.Example(`
+// The waypoint.hcl file
+release {
+	use "nomad-jobspec-canary" {
+		groups = [
+			"app"
+		]
+		fail_deployment = false
+	}
+}
+
+// The app.nomad.tpl file
+job "web" {
+  datacenters = ["dc1"]
+
+  group "app" {
+		network {
+      mode = "bridge"
+      port "http" {
+        to = 80
+      }
+		}
+
+		// Setting a canary in the update stanza indicates a canary deployment
+		update {
+			max_parallel = 1
+			canary       = 1
+			auto_revert  = true
+			auto_promote = false
+			health_check = "task_states"
+		}
+
+		service {
+			name = "app"
+			port = 80
+			connect {
+				sidecar_service {}
+			}
+		}
+
+    task "app" {
+      driver = "docker"
+      config {
+        image = "${artifact.image}:${artifact.tag}"
+				ports  = ["http"]
+      }
+
+      env {
+        %{ for k,v in entrypoint.env ~}
+        ${k} = "${v}"
+        %{ endfor ~}
+
+        // Ensure we set PORT for the URL service. This is only necessary
+        // if we want the URL service to function.
+        PORT = 80
+      }
+    }
+  }
+
+  group "app-gateway" {
+    network {
+      mode = "bridge"
+      port "inbound" {
+        static = 8080
+        to     = 8080
+      }
+    }
+
+    service {
+      name = "gateway"
+      port = "8080"
+
+      connect {
+        gateway {
+          proxy {}
+
+          ingress {
+            listener {
+              port = 8080
+              protocol = "http"
+              service {
+                name  = "app"
+                hosts = [ "*" ]
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  meta = {
+    // Ensure we set meta for Waypoint to detect the release URL
+    "waypoint.hashicorp.com/release_url" = "http://app.ingress.dc1.consul:8080"
+  }
+}
+`)
+
+	doc.SetField(
+		"groups",
+		"List of task group names which are to be promoted.",
+	)
+
+	doc.SetField(
+		"fail_deployment",
+		"If true, fails the canary deployment.",
```

Comment:
> Is it that if it's true, instead of promoting the deployment for release, it fails the deployment by marking it as bad in Nomad? 

^ correct! But agreed that it could be articulated better! I intend to update it to "If true, marks the deployment as failed." - stolen from the Nomad CLI docs! 😄 

https://www.nomadproject.io/docs/commands/deployment/fail#command-deployment-fail